Assets/02.Scripts/02-9. System/CSVReader.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using UnityEngine;
+
+public class CSVReader
+{
+    static string SPLIT_RE = @",(?=(?:[^""]*""[^""]*"")*(?![^""]*""))";
+    static string LINE_SPLIT_RE = @"\r\n|\n\r|\n|\r";
+    static char[] TRIM_CHARS = { '\"' };
+
+    public static List<Dictionary<string, object>> Read(string file)
+    {
+        var list = new List<Dictionary<string, object>>();
+        TextAsset data = Resources.Load(file) as TextAsset;
+
+        var lines = Regex.Split(data.text, LINE_SPLIT_RE);
+
+        if (lines.Length <= 1) return list;
+
+        var header = Regex.Split(lines[0], SPLIT_RE);
+        for (var i = 1; i < lines.Length; i++)
+        {
+
+            var values = Regex.Split(lines[i], SPLIT_RE);
+            if (values.Length == 0 || values[0] == "") continue;
+
+            var entry = new Dictionary<string, object>();
+            for (var j = 0; j < header.Length && j < values.Length; j++)
```

Comment:
의미없는 무분별한 var 사용은 좋지 않습니다!
```suggestion
            for (int j = 0; j < header.Length && j < values.Length; j++)
```